src/utils.c

```diff
@@ -2154,37 +2158,56 @@ const gchar *utils_resource_dir(GeanyResourceDirType type)
 
 void utils_start_new_geany_instance(const gchar *doc_path)
 {
-	const gchar *command = is_osx_bundle() ? "open" : "geany";
-	gchar *exec_path = g_find_program_in_path(command);
+	const gchar *std_argv[] = { app->executable, "-i", "-c", app->configdir, doc_path, NULL };
+#ifdef MAC_INTEGRATION
+	const gchar *osx_argv[] = { "open", "-n", "-a", "Geany", doc_path, "--args", "-i",
+		"-c", app->configdir, NULL };
+#endif
+	const gchar *const *argv = std_argv;
+	GError *err = NULL;
 
-	if (exec_path)
+	if (!app->executable)
 	{
-		GError *err = NULL;
-		const gchar *argv[6]; // max args + 1
-		gint argc = 0;
-
-		argv[argc++] = exec_path;
-		if (is_osx_bundle())
+		if (utils_is_osx_bundle())
 		{
-			argv[argc++] = "-n";
-			argv[argc++] = "-a";
-			argv[argc++] = "Geany";
-			argv[argc++] = doc_path;
+		#ifdef MAC_INTEGRATION
+			char str[256];
+			size_t size = sizeof str;
+
+			argv = osx_argv;
+			/* --args is since OSX 10.6.x = Darwin 10.x */
+			if (sysctlbyname("kern.osrelease", str, &size, NULL, 0) == -1 || atoi(str) < 10)
+				osx_argv[5] = NULL;
+		#endif
 		}
 		else
 		{
-			argv[argc++] = "-i";
-			argv[argc++] = doc_path;
-		}
-		argv[argc] = NULL;
+			const gchar *prgname = g_get_prgname();
+
+		#ifdef G_OS_UNIX
+			if (prgname && strcmp(prgname, "geany"))
+		#else
+			if (prgname && g_ascii_strcasecmp(prgname, "geany") &&
+				g_ascii_strcasecmp(prgname, "geany.exe"))
```

Comment:
If you just put the `!= 0` in the code, nobody has double-think what it does :).